### PR TITLE
Only record class can have a copy constructor.

### LINKED ISSUE #68345: Copy constructor should not be allowed to omit this initializer call in a type that is not a record
According to https://github.com/dotnet/csharplang/blob/main/proposals/primary-constructors.md#primary-constructors-on-records:

> If an explicit constructor declaration is a "copy constructor" - a constructor that takes a single parameter of the enclosing type - it is not required to call a this initializer, and will not execute the member initializers present in the record declaration.

A copy-constructor in a *record* is allowed to not call `this` initializer. However, the compiler also allows it for non-record type with a primary constructor:

```C#
    public class D(int X)
    {
        protected D(D other)
        {
            Console.Write("copy ctor");
        }

        protected D(string other) // error CS8862: A constructor declared in a type with parameter list must have 'this' constructor initializer.
        {
            Console.Write("not a copy ctor");
        }
    }
```

CS8862 should be reported for the copy constructor as well in this case.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #68345.

Relates to https://github.com/dotnet/roslyn/issues/65697

### CODE REVIEW

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs [hunk 1]
```diff
@@ -115,7 +115,7 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
 
         internal static bool IsCopyConstructor(Symbol member)
         {
-            if (member is MethodSymbol { ContainingType.IsRecordStruct: false, MethodKind: MethodKind.Constructor } method)
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs:118]: On second thought, isn't this change affecting the behavior for a copy constructor in a record struct? Looking at the [spec](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-10.0/record-structs.md#answered), there is no notion of copy constructor in record structs.

**Comment by Kaifoij** [comment]
# #

**Comment by Kaifoij** [comment]
2
```powershell

```

**Review by cston — APPROVED** [review]

**Pull request merged**